src/mlpack/tests/kernel_svm_test.cpp

```diff
@@ -0,0 +1,184 @@
+      labels[i] = 0;
+    }
+    for (size_t i = points / 2; i < points; ++i)
+    {
+      data.col(i) = g2.Random();
+      labels[i] = 1;
+    }
+
+    // Now train a svm object on it.
+    KernelSVM<> svm(data, labels, 1.0, false, 10);
+
+    // Ensure that the error is close to zero.
+    const double acc = svm.ComputeAccuracy(data, labels);
+    if (acc <= 0.98)
+      continue;
+
+    arma::mat testData(inputSize, points);
+    arma::Row<size_t> testLabels(inputSize);
+
+    // Create a test set.
+    for (size_t i = 0; i < points / 2; ++i)
+    {
+      data.col(i) = g1.Random();
+      labels[i] = 0;
+    }
+    for (size_t i = points / 2; i < points; ++i)
+    {
+      data.col(i) = g2.Random();
+      labels[i] = 1;
+    }
+
+    // Ensure that the error is close to zero.
+    const double testAcc = svm.ComputeAccuracy(data, labels);
+    if (testAcc >= 0.95)
+    {
+      success = true;
+      break;
+    }
+  }
+
+  BOOST_REQUIRE_EQUAL(success, true);
+}
+
+/**
+ * Test training of kernel svm for two classes on a mnist 4
+ * and 9 dataset.
+ */
+BOOST_AUTO_TEST_CASE(GaussianKernelSVMMnistDataset)
+{
+  arma::mat dataset;
+  dataset.load("mnist_first250_training_4s_and_9s.arm");
+
+  // Normalize each point since these are images.
+  for (size_t i = 0; i < dataset.n_cols; ++i)
+    dataset.col(i) /= norm(dataset.col(i), 2);
+
+  arma::Row<size_t> labels(dataset.n_cols);
+  for (size_t i = 0; i < dataset.n_cols / 2; ++i)
+  {
+      labels[i] = 0;
+  }
+
+  for (size_t i = dataset.n_cols / 2; i < dataset.n_cols; ++i)
+  {
+    labels[i] = 1;
+  }
+
+  bool success = false;
+  for (size_t trial = 0; trial < 5; ++trial)
+  {
+    // Now train a svm object on it.
+    KernelSVM<arma::mat,
+              kernel::GaussianKernel> svm(dataset,
+                                labels, 1.0, true, 10);
+
+    // Ensure that the error is close to zero.
+    const double testAcc = svm.ComputeAccuracy(dataset, labels);
+    if (testAcc >= 0.95)
+    {
+      success = true;
+      break;
+    }
+  }
+
+  BOOST_REQUIRE_EQUAL(success, true);
+}
+
+/**
+ * Test training of kernel svm on concentric circle dataset.
+ */
+BOOST_AUTO_TEST_CASE(ConcentricCircleDataset)
+{
+  // The dataset, which will have three concentric rings in three dimensions.
+  arma::mat dataset;
+
+  // Now, there are 500 points centered at the origin with unit variance.
+  dataset.randn(3, 500);
+  dataset *= 0.05;
+  arma::Row<size_t> labels(dataset.n_cols);
+  for (size_t i = 0; i < dataset.n_cols / 2; ++i)
+  {
+      labels[i] = 0;
+  }
+
+
+  // Take the second 250 points and spread them away from the origin.
+  for (size_t i = 250; i < 500; ++i)
+  {
+    // Push the point away from the origin by 2.
+    const double pointNorm = norm(dataset.col(i), 2);
+
+    dataset(0, i) += 2.0 * (dataset(0, i) / pointNorm);
+    dataset(1, i) += 2.0 * (dataset(1, i) / pointNorm);
+    dataset(2, i) += 2.0 * (dataset(2, i) / pointNorm);
+    labels[i] = 1;
+  }
```

Comment:
Hey @saksham189 I tried to find the problem in my code but the isn't in my code. We are getting this error because it was not able to delete `dataset`. I also tried to run `KernelPCATest` with valgrind and got same error because we are using same dataset in `KernelPCATest`. Do you have any suggestion over this? I think we need to skip this error.

Reply:
We cannot move ahead until memory checks pass. I think it is failing because you are illegally freeing memory. Can you try to find out where this is happening?
Also we seem to be having a lot of memory leaks that should ideally not be happening. 
Take a look here for more details:
http://ci.mlpack.org/job/pull-requests%20mlpack%20memory/5719/valgrindResult/pid=2288559/

Reply:
Hmm, pretty strange these errors were not there in previous builds.

Reply:
There is definitely some strange happing here, I had to kill the memory job after 5 hours because it was using all cores.

It's currently running `/usr/bin/valgrind.bin --tool=memcheck --leak-check=full --track-origins=yes --num-callers=40 --error-exitcode=1 --xml=yes --xml-file=../reports/tests/KernelSVMTest_GaussianKernelSVMMnistDataset.valgrind.xml --log-file=../reports/tests/KernelSVMTest_Gaus`.

Let me know if you need another hand to debug this.

Reply:
@zoq Yeah, Memory builds are talking too much time we have to look into that. Right now waiting for this to finish I have added some changes may be that solves the error.